src/main/java/seedu/address/logic/newcommands/EditCommand.java

```diff
@@ -0,0 +1,497 @@
+                StudentId studentId = absolutePath.getStudentId().get();
+
+                Student studentToEdit = target.getChild(studentId);
+                Student editedStudent = createEditedStudent(studentToEdit, this.editStudentDescriptor);
+                target.deleteChild(studentId);
+                target.addChild(editedStudent.getId(), editedStudent);
+                state.updateList();
+
+                return new CommandResult(MESSAGE_EDIT_STUDENT_SUCCESS);
+
+            } else {
+                throw new CommandException(MESSAGE_INCORRECT_DIRECTORY_ERROR);
+            }
+        } catch (InvalidPathException invalidPathException) {
+            throw new CommandException(ERROR_MESSAGE_INVALID_PATH);
+        } catch (UnsupportedPathOperationException e) {
+            throw new CommandException(ERROR_MESSAGE_UNSUPPORTED_PATH_OPERATION);
+        }
+    }
+
+    /**
+     * Checks if this EditCommand is equal to another object.
+     *
+     * @param other The object to compare with.
+     * @return True if the objects are equal, false otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditCommand)) {
+            return false;
+        }
+
+        EditCommand otherEditCommand = (EditCommand) other;
+        if (this.editStudentDescriptor == null) {
+            return this.relativePath.equals(otherEditCommand.relativePath)
+                    && this.editGroupDescriptor.equals(otherEditCommand.editGroupDescriptor);
+        } else {
+            return this.relativePath.equals(otherEditCommand.relativePath)
+                    && this.editStudentDescriptor.equals(otherEditCommand.editStudentDescriptor);
+        }
+    }
+
+    /**
+     * Returns a string representation of this EditCommand.
+     *
+     * @return A string representation of the object.
+     */
+    @Override
+    public String toString() {
+        if (this.editStudentDescriptor != null) {
+            return new ToStringBuilder(this)
+                    .add("toEdit", this.editStudentDescriptor)
+                    .toString();
+        }
+        if (this.editGroupDescriptor != null) {
+            return new ToStringBuilder(this)
+                    .add("toEdit", this.editGroupDescriptor)
+                    .toString();
+        }
+        return null;
+    }
+
+    /**
+     * Represents the descriptor for editing the details of a student in ProfBook. The descriptor contains fields for
+     * updating the student's name, phone, email, address, and id. It helps to track which fields have been edited by
+     * the user.
+     * An instance of this class is used within the {@code EditCommand} to specify the details to be edited.
+     */
+    public static class EditStudentDescriptor {
```

Comment:
Reccomand storing this in another file? Unless you have a good reason to keep this as a static class